backend/controllers/user.controller.ts

```diff
@@ -65,155 +51,101 @@ export const createNewUser = async (
     });
 
     if (studentExists) {
-      if (studentExists.isVerified) {
-        throw new AppError(
-          AppErrorName.RECORD_EXISTS_ERROR,
-          "User already exists",
-          400,
-          true,
-        );
-      } else if (studentExists.isVerified === false) {
-        await prisma.user.update({
-          where: {
-            email: email,
-          },
-          data: {
-            otp: otp,
-          },
-        });
-
-        const message = {
-          from: "nomansanjari2001@gmail.com",
-          to: email,
-          subject: "Verify OTP - CampusBuddy",
-          html: `<b>${otp}</b>`,
-        };
-
-        await transporter.sendMail(message);
-
-        res.status(100).json({
-          success: false,
-          message: "User not verified, otp sent",
-        });
-      }
+      res.status(400).json({
+        success: false,
+        message: "User already exists",
+      });
     }
 
-    const newUser = await prisma.user.create({
-      data: {
+    const token = jwt.sign(
+      {
+        institutionID: institution.id,
         username: username,
         firstName: firstName,
         lastName: lastName,
         email: email,
         password: password,
-        otp: otp,
-        isVerified: false,
-        institutionId: institution.id,
-        status: false,
-        yearOfStudy: yearOfStudy,
       },
-    });
-
-    const message = {
-      from: "nomansanjari2001@gmail.com",
-      to: email,
-      subject: "Verify OTP - CampusBuddy",
-      html: `<b>${otp}</b>`,
-    };
-
-    await transporter.sendMail(message);
-
-    res.status(200).json({
-      success: true,
-      message: "User created",
-    });
-  } catch (error) {
-    next(error);
-  }
-};
-
-// send new OTP
-export const resendOTP = async (
-  req: Request,
-  res: Response,
-  next: NextFunction,
-) => {
-  try {
-    const { email } = otpRequestSchema.parse(req.body);
-
-    const otp = otpGenerator.generate(6, {
-      upperCaseAlphabets: false,
-      lowerCaseAlphabets: false,
-      specialChars: false,
-    });
-
-    await prisma.user.update({
-      where: {
-        email: email,
+      process.env.JWT_SECRET ?? "testSecret",
```

Comment:
Please use the env validator instead of going straight to the process.

Reply:
Fixed